util/random/rand.go

```diff
@@ -76,3 +81,51 @@ func (r *randstream) XORKeyStream(dst, src []byte) {
 func New() cipher.Stream {
 	return &randstream{}
 }
+
+// READER_BYTES is how many bytes we expect from each source
+const READER_BYTES = 32
+
+type mixedrandstream struct {
+	Readers []io.Reader
+}
+
+func (r *mixedrandstream) XORKeyStream(dst, src []byte) {
+
+	l := len(dst)
+	if len(src) != l {
+		panic("XORKeyStream: mismatched buffer lengths")
+	}
+
+	var b bytes.Buffer
+	var nerr int
+	for _, reader := range r.Readers {
+		buff := make([]byte, READER_BYTES)
+		n, err := io.ReadFull(reader, buff)
+		if err != nil {
+			nerr++
+		}
+		b.Write(buff[:n])
+	}
+
+	// we are ok with few sources being insecure (i.e., providing less than
+	// READER_BYTES bytes), but not all of them
+	if nerr == len(r.Readers) {
+		panic("all readers failed")
+	}
+
+	// create the XOF output, with hash of collected data as seed
+	h := sha512.New()
+	h.Write(b.Bytes())
+	seed := h.Sum(nil)
+	hash := blake2xb.New(seed)
+	hash.XORKeyStream(dst, src)
+}
+
+// NewMixedStream returns a new cipher.Stream that gets random data from the specified
+// readers. If no reader was given, Go's crypto/rand package is used.
```

Comment:
Can you add more comments as to what it does specifically (mixing entropy sources via hashing and then using blake2 to do the XOR)